# -*- coding: utf-8 -*-
"""PR_Assignment_01.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1LHovRgUP-93lZWu1auxy7wm5SK6BG3ep

#Assignment-1
##Pattern Lab - B


---



---


####Rokibul Islam
####011181102
"""

!pip install biopython
from Bio import SeqIO
import sys
import re
import os
import fileinput

import os
PATH = r"/content/drive/My Drive/prLab/dataset"

path = os.listdir(PATH)
files = [x.split('.')[0] for x in path]
print(files)

# Commented out IPython magic to ensure Python compatibility.
# %cd /content/drive/My Drive/prLab/dataset

for f in files:
    out = open(f + "_sp.txt", "w")
    mode = "train"
    classLabel = "1"

    if "test" in f:
      mode = "test"

    if f[0] == "N":
        classLabel = "0"

    for seq_record in SeqIO.parse(f + ".txt", "fasta"):
      id = str(seq_record.id) + "|" + classLabel + "|" + mode
      sequence = str(seq_record.seq)
      out.write(">" + id + "\n" + sequence + "\n")
    out.close()
    print("All good!")

"""#Generating KMER 1-5 for NA/PA files only."""

!pwd

! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_test_sp.txt" --kmer 1 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_test_K1.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_test_sp.txt" --kmer 2 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_test_K2.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_test_sp.txt" --kmer 3 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_test_K3.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_test_sp.txt" --kmer 4 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_test_K4.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_test_sp.txt" --kmer 5 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_test_K5.csv"

! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_test_sp.txt" --kmer 1 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_test_K1.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_test_sp.txt" --kmer 2 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_test_K2.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_test_sp.txt" --kmer 3 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_test_K3.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_test_sp.txt" --kmer 4 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_test_K4.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_test_sp.txt" --kmer 5 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_test_K5.csv"

! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_train_sp.txt" --kmer 1 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_train_K1.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_train_sp.txt" --kmer 2 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_train_K2.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_train_sp.txt" --kmer 3 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_train_K3.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_train_sp.txt" --kmer 4 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_train_K4.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/NA_train_sp.txt" --kmer 5 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/NA_train_K5.csv"

! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_train_sp.txt" --kmer 1 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_train_K1.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_train_sp.txt" --kmer 2 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_train_K2.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_train_sp.txt" --kmer 3 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_train_K3.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_train_sp.txt" --kmer 4 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_train_K4.csv"
! python descnucleotide/RCKmer.py --file "/content/drive/My Drive/prLab/dataset/PA_train_sp.txt" --kmer 5 --normalize --format csv --out "/content/drive/My Drive/prLab/features/Kmer/PA_train_K5.csv"
! echo That was a lotta files

"""#ENAC 5/10
It was at this moment that I thought, "Why don't I just run all the files in a loop?"
"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   python descnucleotide/ENAC.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --slwindow 5 --format csv --out /content/drive/My\ Drive/prLab/features/ENAC/"${files[i]}"_5.csv
#   python descnucleotide/ENAC.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --slwindow 10 --format csv --out /content/drive/My\ Drive/prLab/features/ENAC/"${files[i]}"_10.csv
# done

"""#Binary"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir Binary
# 
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   python iLearn-nucleotide-basic.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method binary --format csv --out /content/drive/My\ Drive/prLab/features/Binary/"${files[i]}"_bin.csv
# done

"""#KSpace 1 3 5 7 9"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir KSpace
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   for((j=1; j<10; j+=2))
#   do 
#     python descnucleotide/CKSNAP.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --gap $j --format csv --out /content/drive/My\ Drive/prLab/features/KSpace/"${files[i]}"_gap$j.csv
#   done
# done

"""#NCP"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir NCP
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   python iLearn-nucleotide-basic.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method NCP --format csv --out /content/drive/My\ Drive/prLab/features/NCP/"${files[i]}"_ncp.csv
# done

"""#EIIP"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir EIIP
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   python iLearn-nucleotide-basic.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method EIIP --format csv --out /content/drive/My\ Drive/prLab/features/EIIP/"${files[i]}"_eiip.csv
# done

"""PseEIIP"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir PseEIIP
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   python iLearn-nucleotide-basic.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method PseEIIP --format csv --out /content/drive/My\ Drive/prLab/features/EIIP/"${files[i]}"_PseEiip.csv
# done

"""#DAC 1-7"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir DAC
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   for((j=1; j<=7; j++))
#   do 
#     python iLearn-nucleotide-acc.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method DAC --type DNA --lag $j --format csv --out /content/drive/My\ Drive/prLab/features/DAC/"${files[i]}"_DAC$j.csv
#   done
# done

"""#TAC 1-7"""

# Commented out IPython magic to ensure Python compatibility.
# %%shell
# cd "/content/drive/My Drive/prLab/features"
# mkdir TAC
# pwd
# cd "/content/drive/My Drive/prLab/iLearn_master"
# files=( "NA_train_sp" "NA_test_sp" "PA_train_sp" "PA_test_sp")
# 
# for ((i=0; i<=3; i++))
# do
#   for((j=1; j<=7; j++))
#   do 
#     python iLearn-nucleotide-acc.py --file /content/drive/My\ Drive/prLab/dataset/"${files[i]}".txt --method TAC --type DNA --lag $j --format csv --out /content/drive/My\ Drive/prLab/features/TAC/"${files[i]}"_TAC$j.csv
#   done
# done